hrm-service/src/contact/contact-data-access.ts

```diff
@@ -68,3 +168,43 @@ export const patch = async (
     return updatedContact;
   });
 };
+
+export const connectToCase = async (
+  accountSid: string,
+  contactId: string,
+  caseId: string,
+): Promise<Contact | undefined> => {
+  return db.task(async connection => {
+    const updatedContact: Contact = await connection.oneOrNone<Contact>(UPDATE_CASEID_BY_ID, {
+      accountSid,
+      contactId,
+      caseId,
+    });
+    return updatedContact;
+  });
+};
+
+export const getById = async (accountSid: string, contactId: string): Promise<Contact> =>
+  db.task(async connection =>
+    connection.oneOrNone<Contact>(selectSingleContactByIdSql('Contacts'), {
+      accountSid,
+      contactId,
+    }),
+  );
+
+export const search = async (
+  accountSid: string,
+  searchParameters: SearchParameters,
+  limit: number,
+  offset: number,
+): Promise<{ rows: Contact[]; count: number }> => {
+  return db.task(async connection => {
+    const searchResults: (Contact & { totalCount: number })[] = await connection.manyOrNone<
+      Contact & { totalCount: number }
+    >(
+      SELECT_CONTACT_SEARCH,
+      searchParametersToQueryParameters(accountSid, searchParameters, limit, offset),
+    );
+    return { rows: searchResults, count: searchResults.length ? searchResults[0].totalCount : 0 };
```

Comment:
`totalCount` is how many records _would_ be returned if we weren't applying an offset & limit, for use in the pagination UI. I borrowed the same trick that we already used in the case query to get this value as a column on the recordset rather than having to run the same query twice